### DISCUSSION TITLE
Change clickoutside node type to Element or Node to allow usage on SVG elements

### DISCUSSION THREAD

**Discussion by vegeta897** [vnphanquang/svelte-put#245] [Ideas]
Right now the `clickoutside` action can only be used on an `HTMLElement` without a type error.

If you ignore the error and use it on, for example, a `<g>` element inside an SVG, it appears to work perfectly. This is great for taking advantage of the precise hit zones of SVG shapes.

I don't think any methods or properties in the `clickoutside` code require the node to specifically be an `HTMLElement`, but maybe I'm missing something.

I propose broadening this type up to `Element` or `Node`. Is this something that could be considered? Thanks!

**Comment by vnphanquang** [comment]
@vegeta897 that totally makes sense. I definitely got into similar situations before but forgot to patch it. Will be available in an upcoming release shortly.

**Comment by vnphanquang** [comment]
Now available in [@svelte-put/clickoutside@3.0.1](https://github.com/vnphanquang/svelte-put/releases/tag/%40svelte-put%2Fclickoutside%403.0.1). Thanks @vegeta897 a lot for raising this. Closing for now but feel free to re-open if necessary :pray:

**Reply by vegeta897** [reply]
Wow, thank you so much for the super fast patch!
